endpoint/domain_filter_test.go

```diff
@@ -1038,3 +1038,123 @@ func TestMatchTargetFilterReturnsProperEmptyVal(t *testing.T) {
+			isConfigured:         true,
+		},
+		{
+			name:                 "EmptyDomainFilter",
+			domainFilter:         []string{},
+			domainExclude:        []string{},
+			expectedDomainFilter: NewDomainFilterWithExclusions([]string{}, []string{}),
+			isConfigured:         false,
+		},
+		{
+			name:                 "RegexDomainExclusionWithoutRegexFilter",
+			regexDomainExclusion: regexp.MustCompile(`test-v1\.3\.example-test\.in`),
+			expectedDomainFilter: NewRegexDomainFilter(nil, regexp.MustCompile(`test-v1\.3\.example-test\.in`)),
+			isConfigured:         true,
+			matchDomain:          "test-v1.3.example-test.in",
+			expectMatch:          false,
+		},
+		{
+			name:                 "RegexDomainFilterWithMultipleDomains",
+			regexDomainFilter:    regexp.MustCompile(`(example\.com|test\.org)`),
+			expectedDomainFilter: NewRegexDomainFilter(regexp.MustCompile(`(example\.com|test\.org)`), nil),
+			isConfigured:         true,
+			matchDomain:          "api.example.com",
+			expectMatch:          true,
+		},
+		{
+			name:                 "RegexDomainFilterWithWildcardPattern",
+			regexDomainFilter:    regexp.MustCompile(`.*\.staging\..*`),
+			expectedDomainFilter: NewRegexDomainFilter(regexp.MustCompile(`.*\.staging\..*`), nil),
+			isConfigured:         true,
+			matchDomain:          "app.staging.example.com",
+			expectMatch:          true,
+		},
+		{
+			name:                 "RegexDomainExclusionWithComplexPattern",
+			regexDomainExclusion: regexp.MustCompile(`^(internal|private)-.*\.example\.com$`),
+			expectedDomainFilter: NewRegexDomainFilter(nil, regexp.MustCompile(`^(internal|private)-.*\.example\.com$`)),
+			isConfigured:         true,
+			matchDomain:          "internal-service.example.com",
+			expectMatch:          false,
+		},
+		{
+			name:                 "RegexFilterAndExclusionBothPresent",
+			regexDomainFilter:    regexp.MustCompile(`.*\.prod\..*`),
+			regexDomainExclusion: regexp.MustCompile(`temp-.*\.prod\..*`),
+			expectedDomainFilter: NewRegexDomainFilter(regexp.MustCompile(`.*\.prod\..*`), regexp.MustCompile(`temp-.*\.prod\..*`)),
+			isConfigured:         true,
+			matchDomain:          "temp-api.prod.example.com",
+			expectMatch:          false,
+		},
+		{
+			name:                 "RegexWithEscapedSpecialChars",
+			regexDomainFilter:    regexp.MustCompile(`test\-api\.v\d+\.example\.com`),
+			expectedDomainFilter: NewRegexDomainFilter(regexp.MustCompile(`test\-api\.v\d+\.example\.com`), nil),
+			isConfigured:         true,
+			matchDomain:          "test-api.v2.example.com",
+			expectMatch:          true,
+		},
+		{
+			name:                 "RegexExclusionWithNumericPattern",
+			regexDomainExclusion: regexp.MustCompile(`\d{3,}-temp\..*`),
+			expectedDomainFilter: NewRegexDomainFilter(nil, regexp.MustCompile(`\d{3,}-temp\..*`)),
+			isConfigured:         true,
+			matchDomain:          "123-temp.example.com",
+			expectMatch:          false,
+		},
+	}
```

Comment:
Maybe add a test with domainFilter, domainExclude, regexDomainFilter & regexDomainExclusion defined and check we get a RegexDomainFilter.

Reply:
I've updated tests

Reply:
example
```
{
			name:                 "RegexDomainFilter with non regex filters ignored",
			regexDomainFilter:    regexp.MustCompile(`example\.com`),
			regexDomainExclusion: regexp.MustCompile(`excluded\.example\.com`),
			domainFilter:         []string{"example.com"},
			domainExclude:        []string{"excluded.example.com"},
			expectedDomainFilter: NewRegexDomainFilter(regexp.MustCompile(`example\.com`), regexp.MustCompile(`excluded\.example\.com`)),
			isConfigured:         true,
		},
```